TYPO3.Flow/Classes/TYPO3/Flow/Property/TypeConverter/DateTimeConverter.php

```diff
@@ -143,7 +143,8 @@ public function convertFrom($source, $targetType, array $convertedChildPropertie
         if ($dateAsString === '') {
             return null;
         }
-        if (ctype_digit($dateAsString) && $configuration === null && (!is_array($source) || !isset($source['dateFormat']))) {
+        $configurationForm = $configuration === null ? null : $configuration->getConfigurationValue('TYPO3\Flow\Property\TypeConverter\DateTimeConverter', self::CONFIGURATION_DATE_FORMAT);
```

Comment:
What does the name `configurationForm` stand for?
And a nitpick: although we didn't replace all occurrences yet we should use the `class` keyword in new code:
```
$configuration->getConfigurationValue(DateTimeConverter::class);
```